Reverse ETL (#3851)
segmentio/segment-docs: src/reverse-etl/bigquery-setup.md

11. Click **Add Key > Create new key**.
12. In the pop-up window, select **JSON** for the key type and click **Create**. The file will download.
13. Copy all the content within the file you just created and downloaded.
14. Navigate to the Segment UI and paste all the credentials you copied from step 13 into the **Enter your credentials** section.
19. Enter your **Data Location**.
20. Click **Test Connection** to test to see if the connection works. If the connection fails, make sure you have the right permissions and credentials and try again.
6. Click **Create Source** if the test connection is successful.

Once you've added BigQuery as a source, you can [add a model](/docs/reverse-etl/reverse-etl/#step-2-add-a-model).

## Constructing your own role or policy
When you construct your own role or policy, Segment needs the following permissions:

Permission | Details
---------- | --------
`bigquery.datasets.create` | This allows Segment to create/manage a `__segment_reverse_etl` dataset for tracking state between syncs.
`bigquery.datasets.get` | This allows Segment to determine if the aforementioned dataset exists

> `bigquery.jobs.create` | This allows Segment to execute queries on any datasets/tables your model query references, in addition to managing our tables for tracking state

Reviewer: ```suggestion
`bigquery.jobs.create` | This allows Segment to execute queries on any datasets/tables your model query references and manage tables that Segment uses for tracking
```
